code/go/0chain.net/smartcontract/dbs/event/miner.go

```diff
@@ -52,6 +54,43 @@ func (edb *EventDb) GetMiner(id string) (*Miner, error) {
 
 }
 
+type MinerQuery struct {
+	gorm.Model
+	MinerID           null.String
+	N2NHost           null.String
+	Host              null.String
+	Port              null.Int
+	Path              null.String
+	PublicKey         null.String
+	ShortName         null.String
+	BuildTag          null.String
+	TotalStaked       state.Balance
+	Delete            null.Bool
+	DelegateWallet    null.String
+	ServiceCharge     null.Float
+	NumberOfDelegates null.Int
+	MinStake          null.Int
+	MaxStake          null.Int
+	LastHealthCheck   null.Int
+	Rewards           null.Int
+	Fees              null.Int
+	Active            null.Bool
+	Longitude         null.Int
+	Latitude          null.Int
+}
+
+func (edb *EventDb) GetMinersWithFiltersAndPagination(filter MinerQuery, offset, limit int) ([]Miner, error) {
+	var miners []Miner
+	query := edb.Get().Debug().Model(&Miner{}).Where(&filter)
+	if offset != -1 {
+		query = query.Offset(offset)
+	}
+	if limit != -1 {
```

Comment:
>=0

Reply:
keeping consistent default value with offset

Reply:
Have a check >= 0 and not != -1
The checks should be logical and not restricted to a value.
Not sure why all comments starting with > are changed to quote
But everywhere add check >= 0

Reply:
done